File: leetcode/easy/1603_parking.py
class ParkingSystem(object):

    def __init__(self, big, medium, small):
        """
        :type big: int
        :type medium: int
        :type small: int
        """
        self.big = big
        self.medium = medium
        self.small = small
        

    def addCar(self, carType):
        """
        :type carType: int
        :rtype: bool
        """
        if carType==1 and self.big>0:
            self.big-=1
            return True
        elif carType==2 and self.medium>0:
            self.medium-=1
            return True
        elif carType==3 and self.small>0:
            self.small-=1
            return True
        else:
            return False


# Your ParkingSystem object will be instantiated and called as such:
# obj = ParkingSystem(big, medium, small)
# param_1 = obj.addCar(carType)

File: leetcode/easy/1603_parking.cpp
#include<iostream>

using namespace std;

class ParkingSystem {
    public:
    int big, medium, small;
public:
    ParkingSystem(int big, int medium, int small) {
        this->big = big;
        this->medium = medium;
        this->small =  small;
    }
    
    bool addCar(int carType) {
        if (carType==1 && big>0 || carType==2 && medium>0 || carType==3 && small>0)
        {
            return false;
        }
        
        if (carType==1)
        {
            big--;
        }
        else if (carType==2)
        {
            medium--;
        }
        else if (carType==3)
        {
            small--;
        }
    }
};

int main() {
    return 0;
}